store/persistent/benchmarks.go

```diff
@@ -37,72 +42,146 @@ func prepare(s store.Storage, size string, b *testing.B) {
 
 	for c := range out {
 		entry := store.MakeIndexEntry(p, protocolID, c.Bytes())
-		err = s.Put(c, entry)
+		_, err = s.Put(c, entry)
 		if err != nil {
-			b.Fatal(err)
+			t.Fatal(err)
 		}
 	}
 	err = <-errOut
 	if err != nil {
-		b.Fatal(err)
+		t.Fatal(err)
 	}
 
 }
 
-func read(s store.Storage, size string, m *metrics, b *testing.B) {
+// readAll reads all of the cids from a file and tries to get it from
+// the persistent storage.
+func readAll(s store.Storage, size string, m *metrics, t *testing.T) {
 	out := make(chan cid.Cid)
 	errOut := make(chan error, 1)
 
 	file, err := os.OpenFile(testDataDir+size+testDataExt, os.O_RDONLY, 0644)
 	if err != nil {
-		b.Fatal(err)
+		t.Fatal(err)
 	}
 	defer file.Close()
 
 	imp := importer.NewCidListImporter(file)
 
-	b.ResetTimer()
 	go imp.Read(context.Background(), out, errOut)
 	for c := range out {
 		now := time.Now()
 		_, found, err := s.Get(c)
 		if err != nil || !found {
-			b.Errorf("cid not found")
+			t.Errorf("cid not found")
 		}
 		m.getTime.add(time.Since(now).Microseconds())
 
 	}
 	err = <-errOut
 	if err != nil {
-		b.Fatal(err)
+		t.Fatal(err)
 	}
 
 }
 
-func BenchSingleGet(s store.StorageFlusher, size string, b *testing.B) {
+// Benchmark the average time per get by all CIDs and the total storage used.
+func BenchReadAll(s store.PersistentStorage, size string, t *testing.T) {
 	m := initMetrics()
-	prepare(s, size, b)
-	read(s, size, m, b)
+	prepare(s, size, t)
+	readAll(s, size, m, t)
 	err := s.Flush()
 	if err != nil {
-		b.Fatal(err)
+		t.Fatal(err)
 	}
 
-	report(s, m, b)
+	report(s, m, true, t)
 }
 
-func BenchParallelGet(s store.StorageFlusher, size string, b *testing.B) {
-	var wg sync.WaitGroup
+// Benchmark single thread get operation
+func BenchCidGet(s store.PersistentStorage, b *testing.B) {
+	cids, err := utils.RandomCids(1)
+	if err != nil {
+		panic(err)
+	}
+	p, _ := peer.Decode("12D3KooWKRyzVWW6ChFjQjK4miCty85Niy48tpPV95XdKu1BcvMA")
+
+	entry := store.MakeIndexEntry(p, protocolID, cids[0].Bytes())
 
-	wg.Add(20)
-	for i := 0; i < 20; i++ {
-		go func(wg *sync.WaitGroup) {
-			defer wg.Done()
-			BenchSingleGet(s, size, b)
+	cids, _ = utils.RandomCids(4096)
+	s.PutMany(cids, entry)
 
-		}(&wg)
+	// Bench average time for a single get
+	b.Run("Get single", func(b *testing.B) {
+		b.ReportAllocs()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, ok, _ := s.Get(cids[i%len(cids)])
+			if !ok {
+				panic("missing cid")
+			}
+		}
+	})
+
+	// Test time to fetch certain amount of requests
+	for testCount := 1024; testCount < 10240; testCount *= 2 {
+		b.Run(fmt.Sprint("Get", testCount), func(b *testing.B) {
+			b.ReportAllocs()
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				for j := 0; j < testCount; j++ {
```

Comment:
It looks like these two loops are the same as doing:
```go
for i := 0; i < b.N * testCount; i++ {
    _, ok, _ := s.Get(cids[i%len(cids)])
}
```
We can revisit the benchmarks later, but let's get these in for now if they are generating meaningful results.